Dplyr glimpse

Holer90 opened the pull request:
Fake pull-request to review the code.


------------------------------------------------
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

nagydavid commented on Dockerfile:
```diff
@@ -2,7 +2,7 @@ FROM quay.io/condaforge/miniforge3
 # i.e. gh_username=myname
-ARG gh_username=pandas-dev
+ARG gh_username=holer90
 ARG pandas_home="/home/pandas"
```
I can comment I will review possibly on Friday

Holer90 replied:
ok

nagydavid replied:
hello

nagydavid replied:
@Holer90 can you pls rebase your commit to the latest and after create a pull request because it is hard to see what is your contribution

nagydavid reviewed (commented):
hello

DeaMariaLeon committed:
DOC Correcting EX02 errors (#51231)

phofl committed:
ENH: Use lazy copy in infer objects (#50428)

topper-123 committed:
TST: suppress deprecation messages in doc tests for Index.is_foo (#51212)

* DOC: suppress deprecation messages in doc tests in Index.is_*

* DOC: fix issue

luke396 committed:
DOC: Add rolling in reference/groupby.rst (#51119)

* Add rolling in reference/groupby.rst

* Add more docstring

* Add see also

* Change See also

* Fix ci

* Fix ci

* Remove substituion

mroeschke committed:
ENH: Allow dt accessor when using ArrowDtype with datetime types (#50954)

* Add methods

* Adjust accessor, add tests

* finished tests

* Whatsnew number

* Use hasattr

* Fix tests and typing

* Fix param

* Fix typing

* Clarify whatsnew

* Only for M

jbrockmendel committed:
REF: remove aggregate_item_by_item (#51222)

jbrockmendel committed:
REF: remove _transform_item_by_item (#51224)

lukemanley committed:
TST: categorical alignment (#51232)

jbrockmendel committed:
REF: simplify aggregate_frame, column-name-pinning (#51226)

* REF: simplify aggregate_frame

* simplify column name-pinning

phofl committed:
DEP: Deprecated pad/backfill for Series/DataFrame (#51221)

jbrockmendel committed:
REF: de-duplicate BusinessDay apply, apply_array (#51220)

jbrockmendel committed:
REF: remove copy keyword from ensure_foo (#51213)

* REF: remove copy keywd from ensure_foo

* mypy fixup

eduardochaves1 committed:
DOC: Additions to month_name & day_name (#51154)

* DOC: Additions to month_name & day_name

* FIX: Removing whitespaces

* doctest +SKIP: because of unavailable locale

* specifying the OS in which the command locale -a works

topper-123 committed:
TYP: more precise types + add type collections (#51234)

TYP: more precise types + add som etype collections

phofl committed:
ENH: Implement arrow support for read_csv with engine=c (#51128)

* ENH: Implement arrow support for read_csv with engine=c

* Adjust docs

* Fix import

* Review

jbrockmendel committed:
BUG: groupby.agg with numba and as_index=False (#51228)

* BUG: groupby.agg with numba and as_index=False

* smaller-diff implementation

* Whatsnew

MarcoGorelli committed:
API: Timestamp and Timedelta .value changing in 2.0 (#50891)

* wip

* wip

* fixup test collection

* divide by zero to debug LOL

* wip

* more fixups

* fixup test unary ops

* ewm

* test offseetes

* are we there yet?

* _value in ujson

* :label: typing

* wip

* advise to use asm8 instead

* fix test missing

* make value property

* un-xfail test

---------

Co-authored-by: MarcoGorelli <>

albuzenet committed:
BUG validate_docstrings.py flake8 PermissionError (#51244)

remove context manager

Co-authored-by: Alex <alex.buzenet.fr@gmail.com>

abonte committed:
DOC: add/fix Timedeltas docstrings (#51229)

* DOC: add/fix Timedeltas docstrings

* remove function from code_checks.sh

phofl committed:
ENH: Add lazy copy to astype (#50802)

Co-authored-by: Joris Van den Bossche <jorisvandenbossche@gmail.com>

MarcoGorelli committed:
BUG: can't round-trip non-nano Timestamp (#51087)

* add test

* default to ns

* wip

* fixup tests

* fixup tests

* :truck: out_reso -> reso; explicit None check

* keep test mixed

* fixup

* wip

* can undo change now

* fixup

* remove debug stmt

* fix logic

* fix precision calculation

* noop

* exhaust match

* use libc log10, reso -> out_reso, periods_per_second

* uncomment on:

* Revert "uncomment on:"

This reverts commit 27fd951e907414cdd7f60b4f38f7060f7cbe7048.

* set unit to ns if None

---------

Co-authored-by: MarcoGorelli <>

DeaMariaLeon committed:
DOC Correct EX02 docstring errors (#51264)

DOC Correct EX02 errors

jbrockmendel committed:
ENH: add unit, as_unit to dt accessor (#51223)

* ENH: add unit, as_unit to dt accessor

* GH ref

lithomas1 committed:
BLD: Try stripping wheels (#51251)

* BLD: Try stripping wheels

* fix syntax?

phofl committed:
BUG: eval and query not working with ea dtypes (#50764)

* BUG: eval and query not working with ea dtypes

* Fix windows build

* Fix

* Fix another bug

* Fix eval

* Fix

* Fix

* Add arrow tests

* Fix pyarrow-less ci

* Add try except

* Add warning

* Adjust warning

* Fix warning

* Fix

* Update test_query_eval.py

* Update pandas/core/computation/eval.py

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

corralien committed:
DOC: Update pct_change documentation (#51262)

* DOC: Update pct_change documentation

Related to SO question: https://stackoverflow.com/questions/75395345/how-to-avoid-bfill-or-ffill-when-calculating-pct-change-with-nans

* DOC: Update pct_change documentation

jbrockmendel committed:
TST: clean up some groupby xfails (#51257)

jbrockmendel committed:
BUG: IntervalArray.shift with invalid NA fill_value (#51258)

* BUG: IntervalArray.shift with invalid NA fill_value

* Update doc/source/whatsnew/v2.0.0.rst

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

jbrockmendel committed:
DEPR: Grouper.ax, Grouper.obj, Grouper.indexer (#51206)

topper-123 committed:
PERF: improve performance of infer_dtype (#51054)

* PERF: improve performance of infer_dtype

* adds GH-number

* version without ABCs

lukemanley committed:
PERF: ArrowExtensionArray.to_numpy(dtype=object) (#51227)

* PERF: ArrowExtensionArray.to_numpy(dtype=object)

* gh refs

cdcadman committed:
Make pandas/io/sql.py work with sqlalchemy 2.0 (#48576)

phofl committed:
DOC: Minor doc cleanups (#51202)

* Clean some mor whatsnew entries

* Add ExcelFile to docs

* Fix typo

* Fix docs

* Add examples

* Add to ignore

albuzenet committed:
DOC fix EX02 algorithms.factorize docstring (#51243)

* fix algorithms.factorize docstring

* remove from exclusion list

* fix dtypes

---------

Co-authored-by: Alex <alex.buzenet.fr@gmail.com>

jbrockmendel committed:
ENH: support reductions for pyarrow temporal types (#50998)

* ENH: support reductions for pyarrow temporal types

* unit check

* lint fixup

phofl committed:
CI: Test numpy 1.24 on 3.11 build (#51006)

* CI: Add complete build with 1.24 numpy

* Change

* Revert "Change"

This reverts commit 993dec471b1b0d38026659cff9ae0c5aef619530.

* Revert "CI: Add complete build with 1.24 numpy"

This reverts commit b58c1da2a9684f7a9a697ae6a52c3c51f9ee68dd.

* Change 3.11 build

* Fix some tests

* Fix

* Fix

* Fix some warnings

* Fix tests

* Add reason

* Fix windows ci

* Fix

* Update ci/deps/actions-311.yaml

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Add warning category

* Update test_frame_subplots.py

* Update test_series.py

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

phofl committed:
Revert "PERF: ArrowExtensionArray.to_numpy(dtype=object)" (#51273)

Revert "PERF: ArrowExtensionArray.to_numpy(dtype=object) (#51227)"

This reverts commit 7ffc0ad1e803513f7a24e351c198fc0308c1cc9f.

attack68 committed:
DOC: update style.ipynb for 2.0 (#50973)

* update style.ipynb for 2.0

* Update doc/source/user_guide/style.ipynb

* Update doc/source/user_guide/style.ipynb

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update doc/source/user_guide/style.ipynb

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update doc/source/user_guide/style.ipynb

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* ensure imports at top of file

---------

Co-authored-by: JHM Darbyshire (iMac) <attack68@users.noreply.github.com>
Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

attack68 committed:
CLN: remove redundant latex options (#50871)

* remove redundant latex options

* remove redundant latex options

* fix up tests and document defaults

* CLN: remove redundant latex generation code

* whats new text fix

* add versionchanged

* Update pandas/core/generic.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

* Update pandas/core/generic.py

* add to version changed

---------

Co-authored-by: JHM Darbyshire (iMac) <attack68@users.noreply.github.com>
Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

jbrockmendel committed:
CLN: simplify groupby wrapping (#51259)

* CLN: simplify groupby wrapping

* mypy fixup

albuzenet committed:
TST: Consolidate tests that raise in groupby (#50749)

mroeschke committed:
CI: Don't test min pyarrow version with multiple python versions (#51272)

phofl committed:
ENH: Implement CoW for convert_dtypes (#51265)

phofl committed:
ENH: Add CoW optimization to interpolate (#51249)

phofl committed:
BUG: replace with inplace not respecting cow (#51278)

mroeschke committed:
CI/TST: Mark test_basic_series_frame_alignment for python engine as flaky (#51219)

Commit:
read_json engine keyword and pyarrow integration (#49249)

jorisvandenbossche committed:
TST: add CoW tests for xs() and get() (#51292)

jorisvandenbossche committed:
TST: enable CoW copy keyword tests for astype/infer_objects (#51293)

phofl committed:
CI: Xfail plotting tests (#51289)

phofl committed:
ENH: Avoid copying unnecessary columns in setitem by splitting blocks for CoW (#51031)

jorisvandenbossche committed:
API / CoW: DataFrame(<dict of Series>, copy=False) constructor now gives lazy copy (#50777)

cdcadman committed:
DOC: Remove reference to mysql fallback support from user guide. (#51275)

Remove reference to mysql fallback support from user guide.

Co-authored-by: Chuck Cadman <charles.cadman@standard.com>

jbrockmendel committed:
API: dont infer dtype for object-dtype groupby reductions (#51205)

* API: dont infer dtype for object-dtype groupby reductions

* GH ref

lukemanley committed:
CI: unpin pyarrow, fix failing test (#51175)

* unpin pyarrow, fix failing test

* cleanup

* handle NaT/NaN

phofl committed:
ENH: Optimize putmask implementation for CoW (#51268)

jorisvandenbossche committed:
API: Series and DataFrame constructors to return shallow copy (i.e. don't share index) from another Series/DataFrame (#50539)

lithomas1 committed:
TST: Test mask/where inplace don't modify views with CoW (#51303)

jbrockmendel committed:
BUG: Groupby.filter with pd.NA (#51255)

lithomas1 committed:
BLD: Build arm64 wheels on CircleCI (#48952)

* REF: Refactor cibuildwheel logic into pyproject.toml

* BLD: Build arm64 wheels on CircleCI

* Update wheels.yml

* Update config.yml

* fix syntax

* fix typo

* Update config.yml

* define parameters

* fix shell script syntax

* try to run less

* Update config.yml

* Update config.yml

* Update config.yml

mroeschke committed:
CLN: Move DeepChainMap to only location used (#51314)

natmokval committed:
DOC: fix EX02 errors in docstrings (#51306)

jbrockmendel committed:
DOC: fix Series.name docstring, closes #42550 (#51310)

DOC: fix Series.name docstring, clsoes #42550

jorisvandenbossche committed:
BUG: avoid StringArray.__setitem__ to mutate the value being set as side-effect (#51299)

* BUG: avoid StringArray.__setitem__ to mutate the value being set as side-effect

* add whatsnew

phofl committed:
ENH: Add CoW optimization for fillna (#51279)

jorisvandenbossche committed:
TST: add CoW test for setitem with Series being set (#51296)

jorisvandenbossche committed:
TST: avoid mutating DataFrame.values in tests (use iloc instead) (#51301)

phofl committed:
TYP: Fix missed copy annotations in concat (#51325)

natmokval committed:
DOC: fix EX02 errors in docstrings II (#51323)

jorisvandenbossche committed:
ENH: Enable lazy copy in merge() for CoW (#51297)

DeaMariaLeon committed:
DOC Fix EX01 errors - added example (#51288)

* DOC Fix EX01 errors - added example

* DOC Fix EX01 issue

jbrockmendel committed:
BUG: Timestamp.replace handle out-of-pydatetime range (#50348)

* BUG: Timestamp.replace handle out-of-pydatetime range

* update test

* fix repr for out-of-pydatetime bounds

* update test

* xfail npdev

* Update pandas/_libs/tslibs/timestamps.pyx

Co-authored-by: Marco Edward Gorelli <33491632+MarcoGorelli@users.noreply.github.com>

---------

Co-authored-by: Marco Edward Gorelli <33491632+MarcoGorelli@users.noreply.github.com>

MarcoGorelli committed:
CI pre-commit autoupdate (#51290)

* wip

* fixup

---------

Co-authored-by: MarcoGorelli <>

lithomas1 committed:
BLD: Fix regex in arm64 wheel builders (#51326)

Update config.yml

natmokval committed:
DOC: fix EX02 errors in docstrings III (#51337)

luke396 committed:
BUG: Fix agg ingore arg/kwargs when given list like func (#50863)

ParfaitG committed:
BUG: iterparse on `read_xml` ignores repeated elements (#51355)

BUG: iterparse on  ignores repeated elements

jbrockmendel committed:
TST: xfail flaky test (#51354)

xfail flaky test

jbrockmendel committed:
BUG: Period('') raising (#51349)

* BUG: Period('') raising

* GH ref

jbrockmendel committed:
BUG: DatetimeArray+DateOffset result unit (#51334)

jbrockmendel committed:
CLN: idiomatic indexing checks (#51360)

CLN: indexing checks

jbrockmendel committed:
ENH: support td64/dt64 in GroupBy.std (#51333)

* ENH: support td64/dt64 in GroupBy.std

* troubleshoot 32bit builds

* fix test

* troubleshoot 32bit builds

* mypy fixup

* troubleshoot CI

jbrockmendel committed:
ENH: Index set operations with sort=True (#51346)

topper-123 committed:
CLN: clean ._hidden_attrs (#51321)

jbrockmendel committed:
CLN: assorted (#51318)

* CLN: assorted

* lint fixup

* pylint fixup

jbrockmendel committed:
BUG: to_datetime with both origin and unit (#51320)

jbrockmendel committed:
REF: avoid try/except in _gotitem (#51311)

topper-123 committed:
DOC: fix some issues (#51286)

Co-authored-by: Patrick Hoefler <61934744+phofl@users.noreply.github.com>

topper-123 committed:
BUG: bug in Index._should_fallback_to_positional (#51241)

* BUG: big in Index._should_fallback_to_positional

* add gh number

* fix perf

lukemanley committed:
PERF: remove categories (#51344)

FlynnOwen committed:
add typehint for None pct_change (#51343)

MarcoGorelli committed:
BUG: can't resample with non-nano dateindex, out-of-nanosecond-bounds (#51274)

* fix resample non-nano out-of-nano-bounds

* use periods_per_second and abbrev_to_npy_unit

* autotyping

* require without freq.nanos

---------

Co-authored-by: MarcoGorelli <>

MarcoGorelli committed:
Datetime parsing (PDEP-4): allow mixture of ISO formatted strings (#50939)

* allow format iso8601

* fixup tests

* :label: typing

* remove duplicate code

* improve message, use if-statement

* note that exact has no effect if format=iso8601

* point to format=ISO8601 in error message

* allow format="mixed"

* link to iso wiki page

* minor fixups

* double backticks -> single, suggest passing format

* use format=mixed instead of apply in example;

---------

Co-authored-by: MarcoGorelli <>
Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

phofl committed:
CLN: Remove default for columns in pivot (#51367)

* CLN: Remove default for columns in pivot

* CLN: Remove default for columns in pivot

MarcoGorelli committed:
DOC remove outdated instructions to delete branch (#51329)

phofl committed:
PERF: Fix performance regression in get_loc of IntervalIndex (#51339)

Holer90 committed:
added pytest test cases to glimpse

MarcoGorelli committed:
CI fixup pylint (#51381)

ci fixup pylint

Co-authored-by: MarcoGorelli <>

jbrockmendel committed:
BUG: GroupBy.quantile with datetimelike and NaT (#51373)

* BUG: GroupBy.quantile with datetimelike and NaT

* GH refs

* mypy, 32bit fixups

phofl committed:
TST: Add test for set_index not mutating index when parent is mutated (#51324)

phofl committed:
ENH: Avoid copy when possible in merge (#51327)

phofl committed:
ENH: Do unique check before computing indexer in reindex (#51304)

* ENH: Do unique check before computing indexer in reindex

* Refactor

MarcoGorelli committed:
STYLE upgrade black formatter (#51348)

upgrade black formatter

Co-authored-by: MarcoGorelli <>

lithomas1 committed:
BLD: Fix uploading of aarch64 wheels (#51358)

* BLD: Fix uploading of aarch64 wheels

* Update config.yml

* Update config.yml

* Update config.yml

* Update config.yml

* Update config.yml

* fix style

* Update config.yml

* Update config.yml

* style

phofl committed:
CI: Pin matplotlib to < 3.7.0 (#51393)

jorisvandenbossche committed:
BUG: ensure reindex / getitem to select columns properly copies data for extension dtypes (#51197)

phofl committed:
CI: Pin openpyxl to < 3.1.1 (#51391)

natmokval committed:
DOC: fix EX02 errors in docstrings (#51404)

DOC: fix EX02 errors in docstrings V

phofl committed:
CI: Fix windows build failures (#51405)

DeaMariaLeon committed:
DOC: Fix EX01 issues in docstrings (#51387)

* DOC Fix EX01 issues in docstrings

* DOC Fix EX01 errors in docstrings

phofl committed:
ENH: Add support for Index.min/max with arrow string (#51401)

phofl committed:
BUG: Arithmetic inplace ops not respecting CoW (#51403)

jbrockmendel committed:
API: Series[bytes].astype(str) behavior (#49658)

* API: Series[bytes].astype(str) behavior

* choose Series behavior

* use .decode

natmokval committed:
DOC: list supported float dtypes I (#50738)

rhshadrach committed:
CLN: Simplify agg_list_like (#51396)

mroeschke committed:
DOC: Add PyArrow user guide (#51371)

* DOC: Add PyArrow user guide

* Address review

* Use code block

jbrockmendel committed:
CLN: assorted comments (#51410)

phofl committed:
PERF: Fix performance regression due to CoW ref tracking (#51390)

phofl committed:
ENH: Add lazy copy to where (#51336)

* ENH: Add lazy copy to where

* Move tests

MarcoGorelli committed:
API deprecate date_parser, add date_format (#51019)

* wip

* fixup

* update user guide

* whatsnew

* gh number

* update user guide;

* whatsnew

* update whatsnew note with date_format enhancement

* make example ipython code-block

* add tests for date_format

* wip add this to read_excel too

* validate within _parser

* minor fixup

* mention other readers in whatsnew, None -> no_default

* Update v2.0.0.rst

* fixup merge conflict resolution

---------

Co-authored-by: MarcoGorelli <>
Co-authored-by: Patrick Hoefler <61934744+phofl@users.noreply.github.com>

lithomas1 committed:
TST: clip inplace doesn't modify views with CoW (#51402)

jbrockmendel committed:
BUG: GroupBy.ohlc with as_index=False (#51413)

* BUG: GroupBy.ohlc with as_index=False

* GH ref

mroeschke committed:
ENH: Allow ArrowDtype(pa.string()) to be compatable with str accessor (#51207)

* Remove __init__ import

* Add base class methods

* Adapt for groupby

* Start adding test pt 1

* Add more tests part 2

* Test pt 3

* More tests

* finish tests

* xfail dask test due to moved path

* Add whatsnew

* Fix import

* Define len

* Address some comments

* address more dask tests

* Revert groupby change

* fix some tests

* Typing

* Undo downstream changes

* Improve error message

jorisvandenbossche committed:
CI/STYLE: run black before ruff in pre-commit (#51416)

phofl committed:
CoW: Ensure that iterrows does not allow mutating parent (#51271)

jorisvandenbossche committed:
TST: add CoW test for update() (#51426)

phofl committed:
ENH: Improve performance for arrow dtypes in monotonic join (#51365)

kkangs0226 committed:
STYLE: Fix errors in doctests (#51356)

* Ignore F821 in setup.cfg

* Fix doctest F721 errors

* Fix indentation

* Remove F821 line

* Restore commented lines

* Revert template string changes

andyjessen committed:
DOC: Fix file extension in docstring of read_sas (#51428)

The ".sas" file extension refers to a program file and not a SAS data set.

kkangs0226 committed:
TEST: cut() with nullable Int64 dtype (#51384)

natmokval committed:
51236 ex02 errors docstrings vi (#51432)

DOC: fix EX02 errors in docstrings VI

jbrockmendel committed:
API: ArrowDtype.type (#51307)

* API: ArrowDtype.type

* handle Decimal, update docstring, update tests

* patch __contains__

* mypy fixup

* mypy fixup

* mypy fixup

* use pc.any(pc.is_nan...

lukemanley committed:
PERF: ArrowExtensionArray.to_numpy(dtype=object) (#51347)

* PERF: ArrowExtensionArray.to_numpy(dtype=object)

* gh refs

* fix

* cleanup / add comment

Dr-Irv committed:
DOC: add instructions for updating environment (#51388)

* DOC: add instructions for updating environment

* Use -r instead of -f in the pip line for upgrading

Co-authored-by: Marco Edward Gorelli <33491632+MarcoGorelli@users.noreply.github.com>

---------

Co-authored-by: Marco Edward Gorelli <33491632+MarcoGorelli@users.noreply.github.com>
Co-authored-by: Patrick Hoefler <61934744+phofl@users.noreply.github.com>

MarcoGorelli committed:
STYLE remove absolufy-imports, use ruff (#51408)

replace supplanted absolufy-imports

Co-authored-by: MarcoGorelli <>

phofl committed:
ENH: Add dict support for read_csv date_format (#51418)

* ENH: Add dict support for read_csv date_format

* Fix converter

* fixup mypy / some docs stuff

* Add gh ref

---------

Co-authored-by: MarcoGorelli <>

phofl committed:
CI: Fix failing ci for arrow list test (#51441)

topper-123 committed:
BUG: DataFrame.to_html(na_rep="-") with non-scalar data (#51412)

* BUG: DataFrame.to_html(na_rep=-) with non-scalar data

* fixes

* fix test

phofl committed:
ENH: Use MaskedEngine for numeric pyarrow dtypes (#51316)

twoertwein committed:
DOC: run typing checks (#51352)

* DOC: run typing checks

* update wording a bit

* reference python setup

* reference pandas CI

* include url to workflows

phofl committed:
PERF: Improve performance for pyarrow to_numpy with na and na_value (#51439)

natmokval committed:
DOC: fix EX02 errors in docstrings (#51453)

DOC: fix EX02 errors in docstrings IV-A

phofl committed:
REF: Refactor reference handling to the BlockManager level for iterrows (#51431)

DeaMariaLeon committed:
DOC Fix EX01 issues in docstrings (#51440)

* DOC Fix EX01 issues in docstrings

* DOC Fix EX01- correct var name+added to example

---------

Co-authored-by: Patrick Hoefler <61934744+phofl@users.noreply.github.com>

jbrockmendel committed:
TST: un-xfail NaT test on npdev (#51455)

* un-xfail for npdev

* more specific

* Update pandas/tests/scalar/test_nat.py

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

jbrockmendel committed:
REF: simplify python_agg_general (#51447)

jbrockmendel committed:
ENH: support addition with pyarrow string dtypes (#51338)

* ENH: support addition with pyarrow string dtypes

* use pc.binary_join_element_wise

* mypy fixup

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

jbrockmendel committed:
REF: move SelectN from core.algorithms (#51460)

mroeschke committed:
BUG: ArrowExtensionArray.mode(dropna=False) not respecting NAs (#50986)

* BUG: ArrowExtensionArray.mode(dropna=False) not respecting NAs

* Fix tests

* remove parameterization

* Fix whatsnew

* Use value_counts

* Remove unneeded xfail

* Dropna after

* Revert "Dropna after"

This reverts commit 1a680a8bb5d67ded97c2295c27016f1ce79d60f2.

* Remove unused request

phofl committed:
CI: Cancel 32 bit when another job is started (#51465)

* CI: Cancel 32 bit when another job is started

* Update .github/workflows/32-bit-linux.yml

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

---------

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

rhshadrach committed:
BUG: DataFrame reductions dtypes on object input (#51335)

* BUG: DataFrame reductions dtypes

* whatsnew

* dtype fixup; whatsnew

* Add test, fix whatsnew

* Add datetime test

* result_dtype.type

* xfail

* type-ignore

phofl committed:
DOC: Fix intendation of versionadded and typo (#51468)

jbrockmendel committed:
BUG: ArrowDtype.construct_from_string round-trip (#50689)

* BUG: ArrowDtype.construct_from_string round-trip

* suggestions

phofl committed:
BUG: Fix date format identification for dict (#51479)

* BUG: Fix date format identification for dict

* Cleanup docs

* Cleanup docs

phofl committed:
DEP: Enforce deprecation of squeeze in read_excel (#51481)

smij720 committed:
BUG: Fix NameError in tab complete warning test (#51485)

tpackard1 committed:
ENH: Add single label to value_counts (#50955)

jorisvandenbossche committed:
TST: parametrize CoW indexing tests for extension (nullable) dtypes (#51208)

phofl committed:
TST: Add tests for clip with CoW (#51492)

natmokval committed:
DOC: fix EX02 errors in docstrings (#51491)

phofl committed:
ENH: Improve ref-tracking for group keys (#51442)

phofl committed:
DOC: Add comment about inplace/copy proposal (#51487)

Commit:
RLS: 2.0.0rc0

datapythonista committed:
Start 2.1.0

mroeschke committed:
CI/TST: Error on PytestUnraisableExceptionWarning instead of using psutil to check open resources (#51443)

mroeschke committed:
TST/CLN: Remove --skip-* flags in favor of -m "not *" (#51490)

jbrockmendel committed:
REF: stricter typing in Manager.take (#51478)

mroeschke committed:
DEPS: Partial Matplotlib 3.7 support (#51470)

loicdiridollou committed:
DOC: clarify "inplace"-ness of DataFrame.setitem (#51328)

mroeschke committed:
CLN: Reset global config state in tests (#51506)

* CLN: Reset global config state in tests

* Add fixture to keep current configs

mroeschke committed:
CLN: Assorted (#51519)

* Remove unused i

* Remove unneeded scope

* Don't modify test param

* Refactor test_pickles

phofl committed:
CI: Run pipelines on 2.0.x branch (#51513)

datapythonista committed:
DOC: Clean up release notes of 2.0.0 (#51509)

jbrockmendel committed:
TST: fix pyarrow xfails for date/time dtypes (#51281)

jbrockmendel committed:
REF: de-duplicate NDFrame.take, remove Manager.take keyword (#51482)

phofl committed:
PERF: Improve performance for ser + range (#51518)

phofl committed:
CLN: Fix up link in docs (#51530)

datapythonista committed:
DOC: Add whatsnew file for 2.1.0 (#51511)

* DOC: Add whatsnew file for 2.1.0

* Fix sections (use minor release template, not patch template)

* Update v2.1.0.rst

* Update v2.1.0.rst

* Update index.rst

---------

Co-authored-by: Patrick Hoefler <61934744+phofl@users.noreply.github.com>

mroeschke committed:
CLN: read_html global importing check (#51505)

DeaMariaLeon committed:
DOC Fix EX01 issues in docstrings (#51533)

jbrockmendel committed:
REF: combine groupby helpers (#51516)

* REF: simplify _apply_to_column_groupbys, rename apply->apply_groupwise

* REF: consolidate self.as_index check

* REF: remove unused _iterate_slices

rmhowe425 committed:
DOC: Updating the timedeltas.seconds docstring for Issue #50969 (#51522)

* Updating the timedeltas.seconds docstring for Issue #50969

* Updating the docstring for Timedelta.seconds

* Removing docstrings on lines 1068-1071 as they're not needed.

* Update pandas/_libs/tslibs/timedeltas.pyx

---------

Co-authored-by: Marco Edward Gorelli <33491632+MarcoGorelli@users.noreply.github.com>

natmokval committed:
DOC: fix EX02 errors in docstrings (#51529)

jbrockmendel committed:
DEPR: deprecate allowing slice in DataFrame.take (#51539)

jbrockmendel committed:
PERF: assert_index_equal with MultiIndex (#51544)

jbrockmendel committed:
ENH: pyarrow dont raise on division by zero (#51541)

phershbe committed:
DOC: fix EX02 docstring errors for three docstrings (#51550)

* updated some dtypes docstrings

* deleted comment

Commit:
STYLE ruff: enable PIE (#51434)

* Ran pre-commit and fixed all errors

* Formatted using black

* Update .pre-commit-config.yaml

Removed the fix arguments.

* Update stata.py

* revert changes to doc/source/user_guide/style.ipynb

* Update test_indexing.py

* Update test_to_records.py and css.py

Changed dtype_mappings and CSS_EXPANSIONS respectively.

* Update css.py

Ran black pre-commit hook.

phofl committed:
DOC: Cleanup CoW dev docs (#51552)

phofl committed:
DOC: Remove 1.0.0 versionadded/versionchanged (#51469)

* DOC: Remove 1.0.0 versionadded/versionchanged

* Fix docstrings

* Fix

mroeschke committed:
TST: Test ArrowExtensionArray with decimal types (#50964)

* TST: Test ArrowExtensionArray with decimal types

* Version compat

* Add other xfails based on min version

* fix test

* fix typo

* another typo

* only for skipna

* Add comment

* Fix

* undo comments

* Bump version condition

* Skip masked indexing engine for decimal

* Some merge stuff

* Remove imaginary test

* Fix condition

* Fix another test

* Update condition

lukemanley committed:
PERF: DataFrame.clip / Series.clip (#51472)

phofl committed:
BUG: transpose not respecting CoW (#51430)

mroeschke committed:
CI/TST: Enable -W error:::pandas in pyproject.toml (#51312)

phofl committed:
DOC: Clean whatsnew links (#51566)

DeaMariaLeon committed:
DOC Fix EX01 issues in docstrings (#51561)

DOC Fix EX01 errors

jbrockmendel committed:
REF: de-duplicate precision_from_unit (#51483)

phofl committed:
ENH: Series.fillna with empty argument making deep copy (#51568)

lukemanley committed:
PERF: DataFrame.first_valid_index/last_valid_index for EA dtypes (#51549)

phofl committed:
CI: Fix failure due to warning (#51573)

phofl committed:
Revert "REF: de-duplicate precision_from_unit" (#51584)

Revert "REF: de-duplicate precision_from_unit (#51483)"

This reverts commit d08d4510db5e79b8e65b35174a3bc3a968bb9189.

natmokval committed:
DOC: fix EX02 errors in docstrings (#51578)

kathleenhang committed:
CI, ENH: Check each minimum dependency is enforced in *.yaml and environment.yml (#51189)

* Add pin_min_versions_to_ci_deps()

* Fix typos

* Remove debug statements

* Update YAML package versions

* Replace files with updated min versions

* Add exclusions + edge cases

* Revert file to original

* Revert environment.yml

* Fix bugs

* Add helper function

* Ran pre-commit run validate-min-versions-in-sync --all-files

* Fix merge issue

* add parameter for custom toml dependency

* Add dummy test files for test_validate_min_versions_in_sync.py

* Add unit test: test_update_yaml_file_version()

* Add dummy test files for edge cases

* Add parametrization to unit test

* Run pre-commit

* Fix tzdata bug

* Fix tzdata issue

* Refactor

* Update ci deps

* Remove pyarrow

* Revert tzdata

* Update pyarrow

* Revert pyarrow

* Add docstrings

* Remove psutil

* Fix file rewrite issue

* Refactor test data

* minor fixups

* fixup

* improve typing

---------

Co-authored-by: MarcoGorelli <>
Co-authored-by: Marco Edward Gorelli <33491632+MarcoGorelli@users.noreply.github.com>

mroeschke committed:
TST/CLN: Remove unnecessary pyarrow version checking (#51545)

* cleanups

* Add back some checks

* Unneeded warning checking

* fix fillna fallback warnings

* Remove tm.assert_produces_warning

DeaMariaLeon committed:
DOC Fix EX01 issues in docstrings (#51593)

DOC Fix EX01 issues plus line too long

akx committed:
CLN/FIX/PERF: Don't buffer entire Stata file into memory (#49228)

* CLN: StataReader: refactor repeated struct.unpack/read calls to helpers

* CLN: StataReader: replace string concatenations with f-strings

* CLN: StataReader: prefix internal state with underscore

* FIX: StataReader: defer opening file to when data is required

* FIX: StataReader: don't buffer entire file into memory unless necessary

Refs #48922

* DOC: Note that StataReaders are context managers

* FIX: StataReader: don't close stream implicitly

* Apply review changes

mroeschke committed:
TST: Make tests less stateful (#51598)

jbrockmendel committed:
BUG: parsing pyarrow dtypes corner case (#51548)

lukemanley committed:
BUG: Series.corr/cov raising with masked dtype (#51422)

futureprogrammer360 committed:
DOC: Add orient=table explanation to read_json doc (#51601)

mroeschke committed:
CLN: Remove read_orc dtype checking (#51604)

liang3zy22 committed:
Fix EX02 error:is_{integer,string}_dtype (#51610)

Dr-Irv committed:
Remove Sequence[str] as type used in DataFrame.to_string() (#47233)

* API: Disallow sets as index and columns argument in DataFrame constructor

* TYP: Remove Sequence[str] from to_string, to_html

* use list[str] for header argument.  Remove dict as index and columns argument for DataFrame.  Define ListLike

* use Axes rather than Sequence[Hashable]

---------

Co-authored-by: Jeff Reback <jeff@reback.net>

timlod committed:
PERF: Bypass chunking/validation logic in StringDtype__from_arrow__ (#47781)

* Bypass chunking/validation logic in __from_arrow__

Instead of converting each chunk to a StringArray after casting to
array and then concatenating, instead use pyarrow to concatenate chunks
and convert to numpy.

Finally, we bypass validation the validation logic by initializing
NDArrayBacked instead of StringArray.

* Handle zero-chunks correctly & convert None to NA

* Add change to whatsnew

* Add to v2 whatsnew

* Add GH issue to comment about validation bypass

* Add # to GH issue

* Move release note to v2.1.0

rhshadrach committed:
BUG: groupby.agg doesn't include grouping columns in result when selected (#51398)

phofl committed:
REGR: MultiIndex.isin with empty iterable raising (#51605)

jbrockmendel committed:
REF: de-duplicate precision_from_unit attempt 2 (#51594)

* Remove also-unraisable raising

* Restore Y/M cases

topper-123 committed:
TST/bug: add test for NaT in Categorical.isin with empty string (#51587)

* TST: add test for NaT in Categorical with empty string

* remove whatsnew entry

phofl committed:
BUG: not broadcasting when setting empty object (#51585)

topper-123 committed:
DOCS: fix category docs (#51579)

* DOC: comparison to CategoricalDtype()

* Add GH-number

* remove whatsnew entry

jbrockmendel committed:
DOC: object_dtype_tzaware_strings.atype(dt64) raises (#51514)

Co-authored-by: Patrick Hoefler <61934744+phofl@users.noreply.github.com>

Kabiirk committed:
WEB: Obfuscating workgroup email addresses to fix Issue #51209 (#51266)

* WEB: Obfuscating workgroup email addresses to fix Issue #51209

* WEB: Fixtrailling whitespace: Obfuscating workgroup email addresses to fix Issue

* WEB: Obfuscated & regex-proof workgroup email addresses

* WEB: refactor suggestions post code review

---------

Co-authored-by: Marc Garcia <garcia.marc@gmail.com>

mroeschke committed:
API: ArrowExtensionArray.value_counts returns pyarrow.int64 type (#51542)

* API: ArrowExtensionArray.value_counts returns pyarrow.int64 type

* Address arrowstringarray

* Address other tests

* Further cleanup

jbrockmendel committed:
BUG: Timestamp.round overflows (#51494)

* BUG: Timestamp.round overflows

* GH ref

* move whatsnew to 2.1.0

luke396 committed:
BUG: Fix GroupBy.idxmin/idxmax retrun wrong type on empty (#51423)

rhshadrach committed:
BUG: Change default of group_keys to True (#51627)

lukemanley committed:
PERF: ArrowExtensionArray.isna when zero nulls or all nulls (#51630)

DeaMariaLeon committed:
DOC trying to improve readability of example (#51634)

* DOC removed parentheses for readability

* minor nit

---------

Co-authored-by: Marco Edward Gorelli <33491632+MarcoGorelli@users.noreply.github.com>

phofl committed:
ENH: DataFrame.fillna making deep copy for dict arg (#51571)

ChristofKaufmann committed:
DOC: Update doc of parquet compression parameter (#51554)

jbrockmendel committed:
PERF: maybe_convert_objects convert_numeric=False (#51517)

jbrockmendel committed:
BUG: TDI divison giving bogus .freq (#51575)

* BUG: TDI divison giving bogus .freq

* GH ref

jbrockmendel committed:
REF: remove iterate_slices (#51622)

lukemanley committed:
BUG: Series.any and Series.all for empty or all-null pyarrow-backed dtypes (#51631)

* BUG: ArrowExtensionArray boolean reductions for empty or all null array

* whatsnew

* clarify whatsnew

mroeschke committed:
ENH: use native filesystem (if available) for read_orc (#51623)

lukemanley committed:
PERF: DataFrame.where for EA dtype mask (#51574)

phofl committed:
ENH: Optimize CoW for fillna with ea dtypes (#51411)

rhshadrach committed:
DOC: Improvements to groupby.rst (#51626)

phofl committed:
CI: Don't run tests when web folder is updated (#51638)

phofl committed:
API / CoW: constructing DataFrame from DataFrame/BlockManager creates lazy copy (#51239)

phofl committed:
ENH: Optimize replace to avoid copying when not necessary (#50918)

* ENH: Optimize replace to avoid copying when not necessary

* ENH: Optimize replace to avoid copying when not necessary

* Fix mypy

* Fix test

* Update for new ref tracking

* Fix inplace argument

* Add cat tests

* Merge main

datapythonista committed:
WEB: Fix blog links (#51649)

datapythonista committed:
DOC: Remove 1.5.4 release notes (#51654)

lithomas1 committed:
WEB: Add code highlighting in PDEPs and render tables correctly (#51458)

datapythonista committed:
WEB: Sort and fix link to under discussions PDEPs (#51656)

lithomas1 committed:
DEPS: Bump Cython to 0.29.33 (#51660)

Holer90 committed:
refined code documentation

Holer90 committed:
removed test-glimpse.py

Holer90 committed:
Merge branch 'main' into dplyr-glimpse

Holer90 committed:
adhered to suggestions by pre-commit
